arabot/modules/urban.py

```diff
@@ -45,52 +46,49 @@ def __init__(self, ara: Ara):
 
     @command(aliases=["ud"], brief="Search term in Urban Dictionary")
     async def urban(self, ctx: Context, *, term: str):
-        if term.lower() == ctx.ara.owner.name.lower():
-            invite = (
-                await ctx.ara.get_guild(676889696302792774).get_unlimited_invite_link()
-                or Embed.Empty
-            )
-            await ctx.send(
-                embed=Embed(description="An awesome guy").set_author(
-                    name=ctx.ara.owner.name.lower(), url=invite
-                )
-            )
-            return
+        for predefined, definition in self.definitions.items():
+            if term.lower() == predefined.lower():
+                await ctx.send(embed=Embed(description=definition).set_author(name=predefined))
+                return
 
-        if term.lower() == ctx.ara.name.lower():
-            invite = (
-                await ctx.ara.get_guild(676889696302792774).get_unlimited_invite_link()
-                or Embed.Empty
-            )
-            await ctx.send(
-                embed=Embed(description="An awesome bot written by an awesome guy").set_author(
-                    name=ctx.ara.name, url=invite
-                )
-            )
-            return
-
-        data = await ctx.ara.session.fetch_json(self.BASE_URL, params={"term": quote_plus(term)})
-        if not (ud := data.get("list")):
-            if ctx.prefix:
+        if not (definitions := await self.fetch_definitions(term)):
+            if ctx.prefix:  # if command was invoked directly by user, not by urban_listener
                 await ctx.send(f"Definition for {bold(term)} not found")
             return
 
-        embed = Embed().set_author(
-            name=term,
-            url=f"https://www.urbandictionary.com/define.php?term={quote_plus(term)}",
-        )
+        embeds = [
+            Embed(
+                description=dsafe(definition["definition"].replace("[", "").replace("]", ""))[
```

Comment:
As discussed, I suggest making this and L68 clearer but otherwise it looks good.